### PR TITLE
Bjorncs/make thread safer

### PR DESCRIPTION
**Pull request by bjorncs**
Ensure that finalizer does not see stale 'state' value.

FYI @baldersheim

### CODE REVIEW

**Review by baldersheim — APPROVED** [review]
👍

File: config/src/main/java/com/yahoo/config/subscription/ConfigSubscriber.java [hunk 1]
```diff
@@ -29,10 +29,11 @@
  */
 public class ConfigSubscriber {
 
-    private Logger log = Logger.getLogger(getClass().getName());
+    private final Logger log = Logger.getLogger(getClass().getName());
```
Comment (vekterli) [inline config/src/main/java/com/yahoo/config/subscription/ConfigSubscriber.java]: Shouldn't this be `static` as well?

**Review by vekterli — COMMENTED** [review]
:+1:

File: config/src/main/java/com/yahoo/config/subscription/ConfigSubscriber.java (hunk 1 above)
Comment (baldersheim) [reply config/src/main/java/com/yahoo/config/subscription/ConfigSubscriber.java]: Agree, then you need to use ConfigSubscriber.class instead of getClass.

**Pull request merged**